libs-scala/resources/src/main/scala/com/digitalasset/resources/ResourceOwner.scala

```diff
@@ -48,13 +48,13 @@ trait ResourceOwner[+A] {
 
 object ResourceOwner {
   def successful[T](value: T): ResourceOwner[T] =
-    new FutureResourceOwner[T](() => Future.successful(value))
+    new FutureResourceOwner(() => Future.successful(value))
 
-  def failed(throwable: Throwable): ResourceOwner[Nothing] =
-    new FutureResourceOwner[Nothing](() => Future.failed(throwable))
+  def failed[T](throwable: Throwable): ResourceOwner[T] =
```

Comment:
Why do you need this? `ResourceOwner` should be covariant now.

Reply:
I see. I would suggest you leave this as is and either:
- hint the compiler of the type you want to fold to (`.fold[ResourceOwner[Config[Extra]]]`), or
- use `map` followed by `getOrElse`

In the first case you would end up with
```scala
  def owner[Extra](
      name: String,
      extraOptions: OptionParser[Config[Extra]] => Unit,
      defaultExtra: Extra,
      args: Seq[String],
  ): ResourceOwner[Config[Extra]] =
    parse(name, extraOptions, defaultExtra, args)
      .fold[ResourceOwner[Config[Extra]]](ResourceOwner.failed(new Config.ConfigParseException))(
        ResourceOwner.successful)
```
and in the second with
```scala
  def owner[Extra](
      name: String,
      extraOptions: OptionParser[Config[Extra]] => Unit,
      defaultExtra: Extra,
      args: Seq[String],
  ): ResourceOwner[Config[Extra]] =
    parse(name, extraOptions, defaultExtra, args)
      .map(ResourceOwner.successful)
      .getOrElse(ResourceOwner.failed(new Config.ConfigParseException))
```

Reply:
Yeah, I was not too happy with this but I think you're right. I'll revert that change.

Reply:
Done.